image_segmentation/pytorch/runtime/training.py

```diff
@@ -0,0 +1,132 @@
+from tqdm import tqdm
+
+import torch
+from torch.optim import Adam, SGD
+from torch.cuda.amp import autocast, GradScaler
```

Comment:
if `.is_cuda_available()`

Reply:
`torch.cuda.*` are no-ops when cuda is not available. I will leave the code as it is